bskylink/src/cache/cache.ts

```diff
@@ -0,0 +1,255 @@
+import {ToolsOzoneSafelinkDefs} from '@atproto/api'
+
+import {type ServiceConfig} from '../config.js'
+import {redirectLogger} from '../logger.js'
+import {OzoneAgent} from './ozoneAgent.js'
+let cacheCursor: string | undefined
+
+export class EventCache {
+  private rules = new Map<string, ToolsOzoneSafelinkDefs.Event>()
+  private cfg: ServiceConfig | undefined = undefined
+
+  constructor() {
+    this.cfg = undefined
```

Comment:
we could just make cfg a required param of the constructor instead of allowing it to be undefined at instantiation. I don't see a case where we would want to instantiate the event cache without a valid config.

```
export class EventCache {
  constructor(private cfg: ServiceConfig){}
  ...
}
```

should be enough to get rid of these lines 

```
  private cfg: ServiceConfig | undefined = undefined
  this.cfg = undefined
  async getConfig(): Promise<ServiceConfig | undefined> {
    return this.cfg
  }
```

and in addition, in the various places you're accessig config with `this.cfg?.prop` can become `this.cfg.prop`